### PR TITLE
WV-2631: Fix Incorrect Info Dates

### PR DESCRIPTION
**Pull request by Tomcariello**
## Description
Fixes # wv-2631

The Black Marble (Annual, 2012 & 2016) and Black Marble - Nighttime Lights only (Annual, 2012 & 2016) layers have incorrect data in the info box.

It says:
Temporal Coverage: 2011-2014
Date Ranges:
2015 - 2014
2011 - 2010

It should be :
Temporal Coverage: 2012-2016
Date Ranges:
2016 - 2016
2012 - 2012

Other layers are similarly corrected, including:
- LIS_Very_High_Resolution_Lightning_Daily_Climatology_LIS_Mean_Flash_Rate
- LIS_Very_High_Resolution_Lightning_Seasonal_Climatology_LIS_Mean_Flash_Rate
- LIS_Very_High_Resolution_Lightning_Monthly_Climatology_LIS_Mean_Flash_Rate
- LIS_Very_High_Resolution_Lightning_Seasonal_Climatology_LIS_Mean_Flash_Rate
- Landsat_WELD_CorrectedReflectance_Bands157_Global_Annual

## How To Reproduce
- Load production WorldView
- Add "black marble annual"; both Normal & Nightime have the same symptoms so select either/both
- In the layer panel, click the "i" icon to View Description
- Note the Temporal Coverage
- Click *View Dates to expand the view
- Note the Date Ranges are incorrect
- Review other layers in the same manner as necessary

## How To Test

- git fetch --all
- git checkout wv-2631-incorrect-info-dates
- npm run build
- Load local WorldView
- Follow the steps above & confirm the dates are now correct.

 
@nasa-gibs/worldview

### CODE REVIEW

File: tasks/build-options/processTemporalLayer.js [hunk 1]
```diff
@@ -47,15 +47,16 @@ async function processTemporalLayer (wvLayer, value) {
         }
         if (end) {
           endDate = moment(end, dateFormat).format('YYYY-MM-DDTHH:mm:ss[Z]')
-          dateRangeEnd.push(endDate)
+          // dateRangeEnd.push(endDate)
```
Comment (victorsh) [inline tasks/build-options/processTemporalLayer.js:50]: remove comment

File: tasks/build-options/processTemporalLayer.js [hunk 2]
```diff
@@ -47,15 +47,16 @@ async function processTemporalLayer (wvLayer, value) {
...
         }
         if (interval !== 'P1D') {
-          endDate = moment(endDate).add(moment.duration(interval))
-          dateRangeEnd.push(endDate)
+          endDate = moment(endDate).add(moment.duration(interval)).format('YYYY-MM-DDTHH:mm:ss[Z]')
+          // dateRangeEnd.push(endDate)
```
Comment (victorsh) [inline tasks/build-options/processTemporalLayer.js:54]: remove comment

File: web/js/modules/date/util.js [hunk 3]
```diff
@@ -363,12 +363,14 @@ export const coverageDateFormatter = (dateType, date, period) => {
       break;
 
     case 'yearly':
-      if (dateType === 'END-DATE') parsedDate.setFullYear(parsedDate.getFullYear() - 1);
+      // if (dateType === 'END-DATE') parsedDate.setFullYear(parsedDate.getFullYear() - 1);
```
Comment (victorsh) [inline web/js/modules/date/util.js:366]: remove comment

File: web/js/modules/date/util.js [hunk 4]
```diff
@@ -363,12 +363,14 @@ export const coverageDateFormatter = (dateType, date, period) => {
...
+      if (dateType === 'END-DATE') parsedDate.setFullYear(parsedDate.getFullYear());
       dateString = moment(parsedDate).format('YYYY');
       break;
 
     case 'monthly':
-      if (dateType === 'END-DATE') parsedDate.setMonth(parsedDate.getMonth() - 1);
+      // if (dateType === 'END-DATE') parsedDate.setMonth(parsedDate.getMonth() - 1);
```
Comment (victorsh) [inline web/js/modules/date/util.js:372]: remove comment

**Review by victorsh — APPROVED** [review]

**Comment by victorsh** [comment]
Other than the commented out code. Everything looks good on my end.

**Comment by Tomcariello** [comment]
@minniewong This update gets the info dates aligned properly by forcing the times to remain in UTC. Black Marble & Weld display imagery as expected. We are still missing imagery on (at least) these layers for December 31. 

- LIS_Very_High_Resolution_Lightning_Daily_Climatology_LIS_Mean_Flash_Rate
- LIS_Very_High_Resolution_Lightning_Seasonal_Climatology_LIS_Mean_Flash_Rate
- LIS_Very_High_Resolution_Lightning_Monthly_Climatology_LIS_Mean_Flash_Rate
-

**Review by ryanweiler92 — APPROVED** [review]

**Review by minniewong — APPROVED** [review]
Looks good!

**Pull request merged**